node/resolvers/search.ts

```diff
@@ -0,0 +1,214 @@
+  geolocationRules: GeolocationRules,
+  addressComponent: google.maps.GeocoderAddressComponent
+) {
+  const geolocationField = geolocationRules[fieldName]
+  address[fieldName] = addressComponent[geolocationField!.valueIn]
+  return address
+}
+
+function getCountry(place: Place) {
+  const countryComponent = place?.address_components?.find(
+    component => component.types.indexOf(AddressType.country) !== -1
+  )
+
+  return countryComponent
+    ? toAlpha3(countryComponent.short_name as ISOAlpha2)
+    : null
+}
+
+export const queries = {
+  getAddressByExternalId: async (
+    _: unknown,
+    args: { id: string },
+    ctx: Context
+  ): Promise<Address> => {
+    const { clients } = ctx
+    const { id } = args
+    const { apiKey } = await clients.apps.getAppSettings(
+      process.env.VTEX_APP_ID
+    )
+
+    const client = new Client({})
+
+    const response = await client.placeDetails({
+      params: {
+        place_id: id,
+        language: Language.pt_BR,
+        key: apiKey,
+      },
+      timeout: 1000,
+    })
+
+    if (response.statusText !== Status.OK) {
+      console.error(response.statusText)
+    }
+
+    const { result: place } = response.data
+
+    const baseAddress: Address = {
+      addressId: '1',
+      addressType: 'residential',
+      city: null,
+      complement: null,
+      country: null,
+      geoCoordinates: [],
+      neighborhood: null,
+      number: null,
+      postalCode: null,
+      receiverName: null,
+      reference: null,
+      state: null,
+      street: null,
+    }
+
+    const country = getCountry(place)
+
+    const rules = countryRules[country!]
+
+    if (!rules) {
+      console.warn(
+        `We don't have geolocation rules for the country: ${country}`
+      )
+      return baseAddress
+    }
+
+    const geolocationRules = rules.geolocation
+
+    function setAddressFields() {
+      const indexedRules = revertRuleIndex(geolocationRules)
+
+      if (!place.address_components) {
+        return {} as Address
+      }
+
+      return place.address_components.reduce<Address>(
+        (updatedAddress: Address, component) => {
+          const checkoutFieldName = getCheckoutFieldName(
+            component.types,
+            indexedRules
+          )
+
+          if (checkoutFieldName) {
+            return setAddressFieldValue(
+              updatedAddress,
+              checkoutFieldName,
+              geolocationRules,
+              component
+            )
+          }
+
+          return updatedAddress
+        },
+        {} as Address
+      )
+    }
+
+    function setGeoCoordinates(updatedAddress: Address) {
+      const {
+        location: { lng, lat },
+      } = place.geometry as AddressGeometry
+
+      updatedAddress.geoCoordinates = [lng, lat]
+
+      return updatedAddress
+    }
+
+    function setCountry(updatedAddress: Address) {
+      updatedAddress.country = country
+      return updatedAddress
+    }
+
+    return pipe(
```

Comment:
can we _not_ use ramda? 😅 we are kinda moving away from using it and prefer using native JS functions (although in this case it doesn't exist yet)